### DISCUSSION TITLE
Why do public packages need authentication?

### DISCUSSION THREAD

**Discussion by meetmangukiya** [community/community#25979] [Actions]
<p>I’ve recently published a public package in github npm registry - <a href="https://github.com/flamy-dev/cornerstoneWebImageLoader/packages/297934" rel="nofollow noopener">https://github.com/flamy-dev/cornerstoneWebImageLoader/packages/297934</a>. I added this to a repo, and the CI was unable to build because it needed authentication for accessing the repo.</p>
<pre><code class="lang-auto">@flamy-dev:registry=https://npm.pkg.github.com⏎
</code></pre>
<p>Added this to <code>.npmrc</code> as well</p>
<p>yarn error message:</p>
<pre><code class="lang-auto">error An unexpected error occurred: "https://npm.pkg.github.com/download/@flamy-dev/cornerstone-web-image-loader/2.1.1/440e376405ba5a0303c61d6d2045e67aac9f257d8f7489a49ef9123438c5ed2e: Request failed \"401 Unauthorized\"".
</code></pre>

**Comment by jcansdale** [comment]
<p>Hi <a class="mention" href="/meetmangukiya">@meetmangukiya</a>,</p>
<p>Unfortunately, you need to use a PAT with the <code>read:packages</code> scope to access packages associated with a public repository.</p>
<p>If you want developers to be able to clone and build your project without generating their own PAT, you can create a <code>.npmrc</code> file at the root of your repository like this:</p>
<pre><code class="lang-auto">@OWNER:registry=https://npm.pkg.github.com
//npm.pkg.github.com/:_authToken="\u003c\u0052\u0045\u0041\u0044\u005f\u0050\u0041\u0043\u004b\u0041\u0047\u0045\u0053\u005f\u0054\u004f\u004b\u0045\u004e\u003e"
</code></pre>
<p><code>@OWNER</code> is your GitHub username and <code>_authToken</code> is a PAT with the <code>read:packages</code> scope. It is encoded so that GitHub won’t automatically delete the PAT when it’s pushed to a public repository.</p>
<p>If you have Docker installed, you can generate the <code>.npmrc</code> file like this:</p>
<pre><code class="lang-auto">docker run jcansdale/gpr encode &lt;READ_PACKAGES_TOKEN&gt;
</code></pre>
<p>I hope that helps!</p>

**Comment by meetmangukiya** [comment]
<p>So, basically publish my own token? However, this still doesn’t allow people to add this package as dependency and be able to build their project without a token, right?</p>

**Comment by jcansdale** [comment, marked as answer]
<p><a class="mention" href="/meetmangukiya">@meetmangukiya</a>,</p>
<blockquote>
<p>However, this still doesn’t allow people to add this package as dependency and be able to build their project without a token, right?</p>
</blockquote>
<p>Right. Packages are only accessible to GitHub accounts via a PAT, they’re not completely public. They’re really geared towards private dependencies not public dependencies (like <a href="http://npmjs.com" rel="nofollow noopener">npmjs.com</a> or <a href="http://NuGet.com" rel="nofollow noopener">NuGet.com</a> is).</p>
<p>You might want to use them for dependencies between repositories, when a package isn’t ready to be published to a public repository yet.</p>
<p>Does that make sense?</p>
